modules/veteran_verification/app/controllers/veteran_verification/v0/keys_controller.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'base64'
+require_dependency 'notary'
+
+module VeteranVerification
+  module V0
+    # KeysController returns a set of public keys as Base64URL encoded PEMs
+    # that can be matched based on the kid (key id) and used to verify
+    # payloads signed by the same Notary in other endpoints.
+    class KeysController < ApplicationController
+      skip_before_action(:authenticate)
+
+      NOTARY = VeteranVerification::Notary.new(Settings.vet_verification.key_path)
+
+      def index
+        render json: {
+          keys: [
+            {
```

Comment:
Making this compliant with the JWKS spec has been difficult, so @nfasulo and I discussed dropping it entirely. Nobody asked us to support JWKS, but we were considering it instead of having something like this `pem` field. In particular, the difficult part is the encoding the `e` and `n` values. They need to be Base64UrlUInt encoded, which is the base64 url encoding of their big-endian representations. 

https://tools.ietf.org/html/rfc7518#section-6.3

In particular, check out the part for the exponent (e):
```
For instance, when representing the value 65537, the octet sequence
to be base64url-encoded MUST consist of the three octets [1, 0, 1];
the resulting representation for this value is "AQAB".
```

And here's more on the definition of Base64UrlUInt: https://tools.ietf.org/html/rfc7518#section-2

I'm not opposed to adding in the `e` and the `n` so that we can also say we are JWK compliant. But want to chat about what a workable conversion for that looks like?

Reply:
Here's a demo of me encoding an e/n value from a key locally. I think this should be similar to the logic we'll use here.

```
2.6.2 :001 > require 'openssl'
 => true
2.6.2 :002 > require 'base64'
 => true
2.6.2 :003 > key = OpenSSL::PKey::RSA.new(2048)
 => #<OpenSSL::PKey::RSA:0x00007ff58c0bce88>
2.6.2 :004 > Base64.urlsafe_encode64(key.e.to_s(2))
 => "AQAB"
2.6.2 :005 > Base64.urlsafe_encode64(key.n.to_s(2))
 => "uPInJ-J1fjf4lU9Nf6dq6BIPI4McOxeL4nTKvQhSZYHQWLaZWKAosl-mPGbsODnfT3y8ueNBrELmePOv6_rDq7x0rGHztmVMlnU2QtBqU-0YP4i50YJmYZt0ElGGowU1AxACT7GrP2OjB5yqPFqVTX6Xhawa9YckYPoZrJWrXG08pr_k38gxOnbxzr-4aqcaE7jmAzA9ZgEkP-vGmDncMzV4SDb6_Pl970o_1UF1BLfdHa49k4keT-TkqxmN7EhMEuQxguKkEE88uWuUgF8iCGSeBIpCcy325RHqFFehxZxd4AOKZi9g-JO533DiD79mEoTgwcLfSiKQs-mLhJyrxQ=="
```

It depends on the [OpenSSL::BN.to_s](https://ruby-doc.org/stdlib-2.6.2/libdoc/openssl/rdoc/OpenSSL/BN.html#method-i-to_s) method to return the binary string representation of those parameters

Reply:
It was right in front of me the whole time! Added the `n` and `e` fields encoded according to the spec.